internal/provider/data_source_hvn_peering_connection.go

```diff
@@ -0,0 +1,103 @@
+package provider
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+)
+
+func dataSourceHvnPeeringConnection() *schema.Resource {
+	return &schema.Resource{
+		Description: "The HVN peering connection data source provides information about an existing network peering between HVNs.",
+		ReadContext: dataSourceHvnPeeringConnectionRead,
+		Timeouts: &schema.ResourceTimeout{
+			Default: &peeringDefaultTimeout,
+		},
+		Schema: map[string]*schema.Schema{
+			// Required inputs
+			"peering_id": {
+				Description:      "The ID of the network peering.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ValidateDiagFunc: validateSlugID,
+			},
+			"hvn_1": {
+				Description: "The unique URL of one of the HVNs being peered.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"hvn_2": {
```

Comment:
Looks like this doesn't get used in the request, so could bump this down to Computed outputs.

Reply:
Actually, changed my mind on this. It's not strictly necessary, but it makes it easier to read the data source and understand what it's connecting.